test/test_torch.py

```diff
@@ -12828,6 +12855,179 @@ def test_function_unwrap_message(self):
+            ("neg", doubles, True, True, 'cuda'),
+            ("reciprocal", doubles, True, True, 'cpu'),
+            ("reciprocal", doubles, False, True, 'cuda'),
+            ("round", doubles, True, True, 'cpu'),
+            ("round", doubles, False, True, 'cuda'),
+            ("rsqrt", positives, True, True, 'cpu'),
+            ("rsqrt", positives, False, True, 'cuda'),
+            ("sin", doubles, True, True, 'cpu'),
+            ("sin", doubles, False, True, 'cuda'),
+            ("sigmoid", doubles, True, True, 'cpu'),
+            ("sigmoid", doubles, False, False, 'cuda'),
+            ("sqrt", doubles, True, True, 'cpu'),
+            ("sqrt", doubles, False, True, 'cuda'),
+            ("tan", doubles, True, True, 'cpu'),
+            ("tan", doubles, False, True, 'cuda'),
+            ("tanh", doubles, True, True, 'cpu'),
+            ("tanh", doubles, False, True, 'cuda'),
+            ("trunc", doubles, True, True, 'cpu'),
+            ("trunc", doubles, False, True, 'cuda')
+        ]
+
+        def _test(fn, inputs, has_input_output_mem_overlap_check,
+                  has_internal_mem_overlap_check, dev):
+            if dev != device:
+                return
+            outplace_fn = getattr(torch, fn)
+            inplace_fn = getattr(torch.Tensor, fn + '_')
+            if has_input_output_mem_overlap_check:
+                self.unary_check_input_output_mem_overlap(self, inputs, sz,
+                                                          outplace_fn)
+            else:
+                with self.assertRaises(AssertionError):
+                    self.unary_check_input_output_mem_overlap(self, inputs, sz,
+                                                              outplace_fn)
+            if has_internal_mem_overlap_check:
+                self.unary_check_internal_mem_overlap(self, inplace_fn,
+                                                      device=dev)
+            else:
+                with self.assertRaises(AssertionError):
+                    self.unary_check_internal_mem_overlap(self, inplace_fn,
+                                                          device=dev)
+
+        for test_case in unary_mem_overlap_cases:
+            _test(*test_case)
+
+    @torchtest.for_all_device_types()
+    def test_binary_op_mem_overlap(self, device):
+        ops = [
+            (torch.add, torch.Tensor.add_),
+            (torch.mul, torch.Tensor.mul_),
+            (torch.sub, torch.Tensor.sub_),
+            (torch.div, torch.Tensor.div_)
+        ]
+        for (out_op, inplace_op) in ops:
+            # internal overlap check for inplace_op
+            self.binary_check_internal_mem_overlap(self, inplace_op,
+                                                   device=device)
+            # input/output overlap check for out op
+            self.binary_check_input_output_mem_overlap(self, out_op, device)
+
+    @torchtest.for_all_device_types()
+    def test_addXXX_mem_overlap(self, device):
```

Comment:
`addcmul`, `addcdiv`, and `lerp` are all ternary pointwise ops: they take three tensors. It would be good to turn this into a test case for ternary ops and reuse it for lerp / lerp_.

Reply:
addressed